File: backend/models/room.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Room struct {
	ID              uint           `gorm:"primaryKey" json:"id"`
	BoardingHouseID uint           `json:"boarding_house_id"`
	BoardingHouse   BoardingHouse  `gorm:"foreignKey:BoardingHouseID" json:"boarding_house"`
	RoomNumber      string         `gorm:"size:20" json:"room_number"`
	Price           float64        `json:"price"`
	Status          string         `gorm:"size:20" json:"status"` // available, occupied, maintenance
	UseDefaultPrice bool           `json:"use_default_price"`
	ImageUrls       StringList     `gorm:"type:text" json:"image_urls"`
	CreatedAt       time.Time      `json:"created_at"`
	UpdatedAt       time.Time      `json:"updated_at"`
	DeletedAt       gorm.DeletedAt `gorm:"index" json:"-"`
}

File: rumatenant/lib/models/room.dart
class Room {
  final int id;
  final int boardingHouseId;
  final String roomNumber;
  final double price;
  final String status;
  final bool useDefaultPrice;
  final List<String> imageUrls;

  Room({
    required this.id,
    required this.boardingHouseId,
    this.roomNumber = '',
    this.price = 0,
    this.status = 'available',
    this.useDefaultPrice = false,
    this.imageUrls = const [],
  });

  factory Room.fromJson(Map<String, dynamic> json) {
    return Room(
      id: json['id'] ?? 0,
      boardingHouseId: json['boarding_house_id'] ?? 0,
      roomNumber: json['room_number'] ?? '',
      price: (json['price'] ?? 0).toDouble(),
      status: json['status'] ?? 'available',
      useDefaultPrice: json['use_default_price'] ?? false,
      imageUrls: json['image_urls'] != null
          ? List<String>.from(json['image_urls'])
          : [],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'boarding_house_id': boardingHouseId,
      'room_number': roomNumber,
      'price': price,
      'status': status,
      'use_default_price': useDefaultPrice,
      'image_urls': imageUrls,
    };
  }
}
